x-pack/plugins/ml/server/plugin.ts

```diff
@@ -87,7 +87,10 @@ export class MlServerPlugin implements Plugin<MlPluginSetup, MlPluginStart, Plug
       navLinkId: PLUGIN_ID,
       app: [PLUGIN_ID, 'kibana'],
       catalogue: [PLUGIN_ID],
-      privileges: null,
+      privileges: {
+        all: admin,
+        read: user,
+      },
       reserved: {
         description: i18n.translate('xpack.ml.feature.reserved.description', {
           defaultMessage:
```

Comment:
The existing (reserved) feature displays this message to the user:

![image](https://user-images.githubusercontent.com/5295965/86979746-10b40880-c150-11ea-9bd9-339458de62f7.png)

Am I correct in saying that, with the changes in this PR, that's not the case anymore?
If so, I believe you can get rid of this entire `reserved` field (along with the corresponding localization messages). Also I think you'll have to update the integration tests. Please correct me if I'm wrong on any of this @legrego.

Also note: making this change moves the feature in the "Space privileges" flyout. It looks like lots of them are out of order, though, so I'm not sure that this matters. Also correct me if I'm wrong @legrego.

Reply:
> If so, I believe you can get rid of this entire reserved field (along with the corresponding localization messages). Also I think you'll have to update the integration tests. Please correct me if I'm wrong on any of this @legrego.

This is a valid question, and a good observation on your part. ML is actually in the process of transitioning to become a "first-class" feature which doesn't rely on reserved roles. _However_, there are existing deployments which are relying on the reserved privileges to grant access to ML, and we can't break those setups for the remainder of 7.x. So until we can make a breaking change, ML needs to support both the reserved and "normal" authorization models.

We updated the UI to allow administrators to use the new set of privileges, but there's nothing stopping them from using the reserved privilege model if they need to do so. We just don't advertise it anymore.

> Also note: making this change moves the feature in the "Space privileges" flyout. It looks like lots of them are out of order, though, so I'm not sure that this matters. Also correct me if I'm wrong @legrego.

The ordering of features is a hot mess, but the move is expected: Features with _only_ reserved privileges are sorted to the bottom of the list, where all other features respect the optional `order` configuration that is specified during feature registration. https://github.com/elastic/kibana/issues/54794 will improve the organization significantly